src/components/Board/BoardShare/BoardShare.component.js

```diff
@@ -67,28 +75,36 @@ const BoardShare = ({
       </DialogTitle>
       <DialogContent className="ShareDialog__content">
         <div className="ShareDialog__Subtitle">
-          <div>{isOwnBoard && <Button
-            color="primary"
-            variant={isPublic ? "outlined" : "contained"}
-            className="ShareDialog__ToggleStatusButton"
-            onClick={publishBoard}
-            >
-              {!isPublic? <FormattedMessage {...messages.publishBoard } /> : <FormattedMessage {...messages.unpublishBoard } />}
-            </Button>
+          <div>{
+            isOwnBoard  
+              ? <Button
+              color="primary"
+              variant={isPublic ? "outlined" : "contained"}
+              className="ShareDialog__ToggleStatusButton"
+              onClick={publishBoard}
+              >
+                {!isPublic? <FormattedMessage {...messages.publishBoard } /> : <FormattedMessage {...messages.unpublishBoard } />}
+              </Button>
+              : <><Alert severity="warning">warning — you need to be registered to share a board</Alert>
+                <Button
+                  color="primary"
+                  variant="contained"
+                  component={Link} to="/login-signup"
+                >LOGIN or SING UP</Button></>
           }
           </div>
         </div>
-
+        {getIsSharable(isPublic,isOwnBoard)}
```

Comment:
Any reason to use a global variable and assign it a new value from inside the JSX expression? Perhaps it would be better to declare `isSharable` right before the React fragment.